tendermint/src/rpc/endpoint/abci_info.rs

```diff
@@ -27,41 +27,58 @@ impl rpc::Response for Response {}
 
 /// ABCI information
 #[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(default)]
 pub struct AbciInfo {
     /// Name of the application
     pub data: String,
 
     /// Version
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub version: Option<String>,
 
     /// Last block height
-    pub last_block_height: block::Height,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub last_block_height: Option<block::Height>,
 
     /// Last app hash for the block
     #[serde(
         serialize_with = "serialize_app_hash",
-        deserialize_with = "parse_app_hash"
+        deserialize_with = "parse_app_hash",
+        skip_serializing_if = "Option::is_none"
     )]
-    pub last_block_app_hash: Hash,
+    pub last_block_app_hash: Option<Hash>,
 }
 
 /// Parse Base64-encoded app hash
-pub(crate) fn parse_app_hash<'de, D>(deserializer: D) -> Result<Hash, D::Error>
+pub(crate) fn parse_app_hash<'de, D>(deserializer: D) -> Result<Option<Hash>, D::Error>
 where
     D: Deserializer<'de>,
 {
     let bytes = base64::decode(String::deserialize(deserializer)?.as_bytes())
         .map_err(|e| D::Error::custom(format!("{}", e)))?;
 
-    Hash::new(hash::Algorithm::Sha256, &bytes).map_err(|e| D::Error::custom(format!("{}", e)))
+    Hash::new(hash::Algorithm::Sha256, &bytes)
+        .map(Some)
+        .map_err(|e| D::Error::custom(format!("{}", e)))
 }
 
 /// Serialize Base64-encoded app hash
-pub(crate) fn serialize_app_hash<S>(hash: &Hash, serializer: S) -> Result<S::Ok, S::Error>
+pub(crate) fn serialize_app_hash<S>(hash: &Option<Hash>, serializer: S) -> Result<S::Ok, S::Error>
 where
     S: Serializer,
 {
-    String::from_utf8(base64::encode(hash.as_bytes()))
+    String::from_utf8(base64::encode(hash.unwrap().as_bytes()))
```

Comment:
Won't this panic if hash is `None`?

Reply:
That's an interesting point. The normal execution in `serde` is that if the parameter is missing or None and you have the `Default` trait implemented, then this serializer function is not run. The `Default` implementation takes over and assumes whatever value is defined there.

Reply:
There is definitely something fishy about this 🤔 I think the method should err if hash is `None` which is possible. At least the type system allows that even if used as intended this won't happen.

Reply:
That's why we have the `skip_serializing_if = "Option::is_none"` added. If hash is `None`, `serde` will not call the function.